File: code/jvm/domain/src/main/java/pt/isel/domain/invitations/Invitation.java
package pt.isel.domain.invitations;

import pt.isel.domain.channels.AccessType;
import pt.isel.domain.channels.Channel;
import pt.isel.domain.users.UserInfo;

import java.time.LocalDateTime;

public record Invitation(
        Long id,
        String token,
        UserInfo createdBy,
        Channel channel,
        AccessType accessType,
        LocalDateTime expiresAt,
        InvitationStatus status
) {
    public Invitation(Long id, String token, UserInfo createdBy, Channel channel, AccessType accessType, LocalDateTime expiresAt) {
        this(id, token, createdBy, channel, accessType, expiresAt, InvitationStatus.PENDING);
    }
}

File: code/jvm/domain/src/main/kotlin/pt/isel/domain/invitations/Invitation.kt
package pt.isel.domain.invitations

import pt.isel.domain.channels.AccessType
import pt.isel.domain.channels.Channel
import pt.isel.domain.users.UserInfo
import java.time.LocalDateTime

data class Invitation(
    val id: Long,
    val token: String,
    val createdBy: UserInfo,
    val channel: Channel,
    val accessType: AccessType,
    val expiresAt: LocalDateTime,
    val status: InvitationStatus = InvitationStatus.PENDING,
)
